rtmp-client/src/main/cpp/librtmp-jni.c

```diff
@@ -37,34 +33,35 @@ Java_net_butterflytv_rtmp_1client_RtmpClient_nativeOpen(JNIEnv* env, jobject thi
     RTMP *rtmp = (RTMP *) rtmpPointer;
    // rtmp = RTMP_Alloc();
     if (rtmp == NULL) {
-        return -1;
+        throwIllegalStateException(env, "RTMP open called without allocating rtmp object");
+        return RTMP_ERROR_IGNORED;
     }
 
     RTMP_Init(rtmp);
     rtmp->Link.receiveTimeoutInMs = receiveTimeoutInMs;
     rtmp->Link.sendTimeoutInMs = sendTimeoutInMs;
-    int ret = RTMP_SetupURL(rtmp, url);
+    RTMPResult ret = RTMP_SetupURL(rtmp, url);
 
-    if (!ret) {
+    if (ret != RTMP_SUCCESS) {
         RTMP_Free(rtmp);
-        return -2;
+        return ret;
     }
     if (isPublishMode) {
         RTMP_EnableWrite(rtmp);
     }
 
     ret = RTMP_Connect(rtmp, NULL);
-    if (!ret) {
+    if (ret != RTMP_SUCCESS) {
         RTMP_Free(rtmp);
-        return -3;
+        return ret;
     }
     ret = RTMP_ConnectStream(rtmp, 0);
 
-    if (!ret) {
-        return -4;
+    if (ret != RTMP_SUCCESS) {
```

Comment:
Not sure, but is an `RTMP_Free` call needed here?

Reply:
I've added it.